File: 392.Is_Subsequence/main.ts
class Solution {
     public isSubsequence(s: string, t: string): boolean {
         let sIndex = 0, tIndex = 0;
         while (sIndex < s.length && tIndex < t.length) {
             if (s[sIndex] === t[tIndex]) {
                 sIndex++;
             }
             tIndex++;
         }
         return sIndex === s.length;
     }
 }
 
 const solution = new Solution();
 const s = "abc";
 const t = "ahbgdc";
 console.log(solution.isSubsequence(s, t)); // Output: true
 

File: 392.Is_Subsequence/main.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution
{
public:
     bool isSubsequence(string s, string t)
     {
          int sIndex = 0, tIndex = 0;
          while (sIndex < s.size() && tIndex < t.size())
          {
               if (s[sIndex] == t[tIndex])
               {
                    sIndex++;
               }
               tIndex++;
          }
          return sIndex == s.size();
     }
};

int main()
{
     Solution solution;
     string s = "abc";
     string t = "ahbgdc";
     cout << (solution.isSubsequence(s, t) ? "true" : "false") << endl; // Output: true
     return 0;
}
